### PR TITLE
Replace shared mocks in incoming transaction controller tests

### PR DESCRIPTION
**Pull request by Gudahtt**
The shared mocks used previously in the incoming transaction controller tests have been replaced with functions that can generate a new mock for each test.

We should avoid ever sharing mocks between tests. It's quite easy for a mock to get accidentally mutated or not correctly "reset" for the next test, leading to test inter-dependencies and misleading results.

In particular, it is unsafe to share a `sinon` fake (e.g. a spy or stub) because they can't be fully reset between tests. Or at least it's
difficult to reset them property, and it can't be done while also following their recommendations for preventing memory leaks.

The spy API and all related state can be reset with `resetHistory`, which can be called between each test. However `sinon` also recommends calling `restore` after each test, and this will cause `sinon` to drop its internal reference to the fake object, meaning any subsequent call to `resetHistory` would fail. This is intentional, as it's required to prevent memory from building up during the test run, but it also means that sharing `sinon` fakes is particularly difficult to do safely.

Instead we should never share mocks in the first place, which has other benefits anyway.

This was discovered while writing tests for #9583. I mistakenly believed that `sinon.restore()` would reset the spy state, and this was responsible for many hours of debugging test failures.

### CODE REVIEW

**Review by whymarrh — APPROVED** [review]
LGTM!

File: test/unit/app/controllers/incoming-transactions-test.js [hunk 1]
```diff
@@ -2,38 +2,51 @@ import assert from 'assert'
...
-      [ROPSTEN]: 1,
-      [RINKEBY]: 2,
-      [KOVAN]: 3,
-      [GOERLI]: 5,
-      [MAINNET]: 4,
+      [GOERLI]: 1,
+      [KOVAN]: 2,
+      [MAINNET]: 3,
+      [RINKEBY]: 5,
+      [ROPSTEN]: 4,
```
Comment (rekmarks) [inline test/unit/app/controllers/incoming-transactions-test.js:43]: I noticed that the values for the networks differ from the mocks in `getNonEmptyInitStateWithFakeNetworkState`, if that matters.

Comment (Gudahtt) [reply test/unit/app/controllers/incoming-transactions-test.js:43]: Ah, I don't think that matters. These are just arbitrary numbers to represent the last block fetched, so they don't need to be the same between mocks. I'll also be replacing `getNonEmptyInitStateWithFakeNetworkState` altogether in a subsequent PR (the one after the next one)

Comment (rekmarks) [reply test/unit/app/controllers/incoming-transactions-test.js:43]: Ah, I see. Never mind then! 👍

**Pull request merged**